BootloaderCommonPkg/Library/Ext23Lib/ExtLib.c

```diff
@@ -54,11 +58,36 @@ ExtInitFileSystem (
   PrivateData->LastBlock     = PartBlockDev->BlockDevice[SwPart].LastBlock;
   PrivateData->BlockSize     = PartBlockDev->BlockInfo.BlockSize;
 
+  // verify FS format by opening root: superblock will be validated
+  OpenFile = (OPEN_FILE *)AllocateZeroPool (sizeof (OPEN_FILE));
+  if (OpenFile == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    goto Error;
+  }
+
+  OpenFile->FileDevData = PrivateData;
+  Status = Ext2fsOpen (Root, OpenFile);

```

Comment:
I guess Ext2fsOpen might be too much for simple verification.   
By looking into the code, I believe ReadSBlock() itself should be good enough for the basic check.
Can we have a short path to just do ReadSBlock to check EXT2 ?

Reply:
Agree with Maurice. Maybe partial code of ReadSBlock() would be enough.
```
if (Ext2Fs.Ext2FsMagic != E2FS_MAGIC) { }
if (Ext2Fs.Ext2FsRev > E2FS_REV1 || ...) { }
```

Reply:
let me rework the patch to partially reuse ReadSBlock.

Reply:
Thank you, Stanley!  :)

Reply:
A new Ext2SbValidate() is implemented for both ExtInitFileSystem() and file open